// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTPeerObject";
option optimize_for = LITE_RUNTIME;

import public "Nym.proto";
import public "PeerEnums.proto";
import public "PeerReply.proto";
import public "PeerRequest.proto";
import public "Purse.proto";

message PeerObject
{
    optional uint32 version = 1;
    optional PeerObjectType type = 2;
    optional string otmessage = 3;
    optional PeerRequest otrequest = 4;
    optional PeerReply otreply = 5;
    optional Nym nym = 6;
    optional string otpayment = 7;
    optional Purse purse = 8;
}
